libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -644,10 +645,16 @@ bool WaveClip::Paste(double t0, const WaveClip &other)
 
    Transaction transaction{ *this };
 
+   const auto otherStretchRatio = other.GetStretchRatio();
+   if (GetSequenceSamplesCount() == sampleCount{0})
+      // This is empty: use stretch ratio of other clip to avoid unnecessary stretching.
+      ApplyStretchRatio(otherStretchRatio, [](double) {});
+   const auto thisStretchRatio = GetStretchRatio();
+   const bool clipNeedsStretching =
+      fabs(thisStretchRatio - other.GetStretchRatio()) > 1e-6;
```

Comment:
Could there be a smarter choice of tolerance?  Maybe calculate whether the applied stretch times duration makes more than `1 / other.GetRate()`

Reply:
This check is meant to ignore numeric errors, which in theory could happen if you do the following: copy a clip, change project tempo multiple times, and paste the clip back. The clipboard clip will have undergone only one project tempo change, at the moment of paste, while the clip that stayed in the project, several.
Nevertheless, I just tested that, and both ratios still end up the exact same. So I'll exchange this for pure equality.

Reply:
> This check is meant to ignore numeric errors, which in theory could happen if you do the following: copy a clip, change project tempo multiple times, and paste the clip back. The clipboard clip will have undergone only one project tempo change, at the moment of paste, while the clip that stayed in the project, several. Nevertheless, I just tested that, and both ratios still end up the exact same. So I'll exchange this for pure equality.

Don't do that.  Never compare results of floating point computations for equality.